core/arch/proc_shared.c

```diff
@@ -133,6 +133,11 @@ proc_init(void)
         global_heap_free(buf, PAGE_SIZE HEAPACCT(ACCT_OTHER));
         os_close(cpuinfo);
     }
+    CLIENT_ASSERT(dcache_zva_size > 0, "invalid dcache block size for zeroing");
```

Comment:
Will this fail in the DR_HOST_NOT_TARGET case? Do we want to maybe use a default to prevent this and other failures due to not getting the `dcache_zva_size`?

Reply:
> Will this fail in the DR_HOST_NOT_TARGET case?

Yes, it's failing:
```
aarch64-on-x86: 3 tests passed, **** 5 tests failed: ****
        tool.drcacheoff.raw2trace_unit_tests 
        code_api|api.ir 
        code_api|api.ir_negative 
        code_api|tool.drcacheoff.altbindir 
        code_api|api.ir-static
```
What needs to be done to run aarch64-on-x86? Is it just a case of setting `dcache_zva_size` to the host's `cache_line_size`?

Reply:
@derekbruening FYI

We discussed an API to set `dcache_zva_size` to a given value, that clients can use if they have that information. E.g. drcachesim can set it based on the value of `TRACE_MARKER_TYPE_CACHE_LINE_SIZE`.

At init time, it would be reasonable to set it to the host's cache line size.

Reply:
> At init time, it would be reasonable to set it to the host's cache line size.

I've committed a fix for the failing `aarch64-on-x86` tests: `tool.drcacheoff.raw2trace_unit_tests` and `tool.drcacheoff.altbindir` now pass. The fix simply sets `dcache_zva_size = cache_line_size` if `DR_HOST_NOT_TARGET`.

`api.ir` fails because the hard-coded size in `ir_aarch64.expect` is `[64byte]` while the runtime value is `[32byte]`
But the runtime value is wrong for my x86 box:
```
$ cat /sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size 
64
$ 

$ getconf LEVEL1_DCACHE_LINESIZE
64
asshas01@man-06:~$
```
So I would expect `get_processor_specific_info()` in `core/arch/x86/proc.c` to also set `cache_line_size` to 64.

My x86's `/proc/cpuinfo` says:
```
processor       : 55
vendor_id       : GenuineIntel
cpu family      : 6
model           : 79
model name      : Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz
. . .
cache size      : 35840 KB
. . .
clflush size    : 64
cache_alignment : 64
```

Reply:
> `api.ir` fails because the hard-coded size in `ir_aarch64.expect` is `[64byte]` while the runtime value is `[32byte]` But the runtime value is wrong for my x86 box:
> ...
> So I would expect `get_processor_specific_info()` in `core/arch/x86/proc.c` to also set `cache_line_size` to 64.

I would guess that for host != target regular DR is not used and this is just the drdecode library, which probably does not run all the regular managed-mode initialization code and so never runs the proc.c code.  Is there *any* initialization for a static library version of drdecode?  The API does not require any init call IIRC.

Reply:
> probably does not run all the regular managed-mode initialization code and so never runs the proc.c code.

Yes, `get_cache_line_size()` and `get_dcache_zero_blk_size()` return false because `DR_HOST_NOT_TARGET`, and so `CTR_EL0` or `DCZID_EL0` can't be read.

Unlike x86, the smallest line size for AArch64 is 64. The latest commit sets `dcache_zva_size` and `cache_line_size` to 64 bytes for `DR_HOST_NOT_TARGET`.

The `api.ir-static` tests are are still failing. Not sure why the base disp doesn't appear at all in test output:
```
$ diff -y suite/tests/ir_aarch64.expect suite/tests/ir_aarch64.expect-out | grep \| | head -3
dc_zva  -> (%x0)[64byte]                                      | dc_zva  -> (%x0)
dc_zva  -> (%x1)[64byte]                                      | dc_zva  -> (%x1)
dc_zva  -> (%x2)[64byte]                                      | dc_zva  -> (%x2)
$
```

Reply:
You mean it's not printing the size `[64byte]`?  Some kind of disassembly mask/option difference?